src/bentoml/_internal/server/runner_app.py

```diff
@@ -311,6 +330,27 @@ async def _request_handler(request: Request) -> Response:
                         "Server": server_str,
                     },
                 )
+            from starlette.responses import StreamingResponse
+
+            if runner_method.config.is_stream:
+
+                async def streamer() -> t.AsyncGenerator[bytes, None]:
+                    """
+                    Extract Data from a AsyncGenerator[Payload, None]
+                    """
+                    async for p in payload:
+                        yield pickle.dumps(p)
```

Comment:
I have a few concerns about this approach:

1. Wouldn't this cause double pickling which will affect the performance?
2. Can the aiohttp client split the chunks correctly? How does it know the boundaries?

Reply:
1. Yes there's double pickling, it is similar to how we are handling `multiple_outputs`. I was doing this to align with how we currently pass payloads across the wire. 

2. I've tested with 1-100MB worth of strings, the aiohttp server will chunk them and on the client side, it is handled with this [logic]. (https://github.com/bentoml/BentoML/pull/4083/commits/7f568dc276911de38e8ecbd404bc8a3302929ed5) The boundaries is handled by library following this [protocol](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding#chunked_encoding)

Reply:
> * Would we only allow user to yield data of one type, or multiple?

> * How would the IO descriptor looks like for streaming of multiple data type

I assume we only support text stream, but this approach seems to allow `yield picklable_object` in the runnable method as well.

> 2\. The boundaries is handled by library following this [protocol](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding#chunked_encoding)

TIL, so it can handle newlines in the serialized form correctly?

Reply:
> I assume we only support text stream, but this approach seems to allow yield picklable_object in the runnable method as well.

Actually user can yield anything, image, nd.array, json, and this approach allows us to generalize that as long as we use the right descriptor and implement streaming for them.

> TIL, so it can handle newlines in the serialized form correctly?
yeap, i can confirm that!

Reply:
fwiw we might want to do a/b testing on this one day

Reply:
you mean a/b test for the additional payload and see the overhead?

Reply:
Anyway, the internal exchange format won't affect the user interface, so LGTM and we can improve the performance(if it has issues) in the future.